kafka-client/src/main/java/dev/responsive/kafka/api/async/internals/contexts/StreamThreadProcessorContext.java

```diff
@@ -126,6 +127,11 @@ public void prepareToFinalizeEvent(final AsyncEvent event) {
     originalContext.setCurrentNode(asyncProcessorNode);
   }
 
+  public void afterFinalizeEvent() {
```

Comment:
nice catch -- but is this correct? Shouldn't we actually stash the current record context and node when we first set it, then set it back to this at the end rather than null?

Reply:
I have an expansion of the integration test that adds stuff before and after the async processor. Was going to wait to open a PR since it's obviously broken right now but I can open it so you can copy/pull in the new topology and test this

Reply:
oof you're right. I'll fix that. I hit this during punctuations so just solved for that case.